### ISSUE TITLE
Everything blinks when changing the wallpaper

### ISSUE THREAD

**Issue by xyproto** [swaywm/sway#3691]
Hi,

I wrote a utility for being able to set Timed Wallpapers (wallpaper slideshows, with crossfade between images) that also works for Wayland/Sway.

This works nicely, but when the wallpaper is changed, everything blinks.

Steps to reproduce:
1. Install the [Mojave timed wallpaper](https://github.com/japamax/gnome-mojave-timed-wallpaper), either from GitHub or from the AUR package named `gnome-mojave-timed-wallpaper` (if you use Arch Linux).
2. Install the [monitor tools](https://github.com/xyproto/monitor), either from GitHub or from the AUR package named `monitor` (if you use Arch Linux).
3. Run this command `settimed mojave-timed`.

Then just wait a bit, and everything will blink every few minutes.

The command being run for setting the wallpaper looks similar to this:

    swaymsg 'output "*" background /usr/share/backgrounds/gnome/mojave/mojave_dynamic-1320.jpg fill'

This is a feature request for the background image to be switched smoothly, without all the windows blinking.

I'm using `sway version 1.0-beta.1-245-gdbf8e1ce (Nov 27 2018, branch 'master')`.

Thanks for creating Sway!

**Comment by emersion** [comment]
This has been fixed in master.

**Comment by xyproto** [comment]
Just installed and tested with master.

The windows have stopped blinking, but the background wallpaper still blinks by changing the wallpaper to gray before changing it to the desired image.

**Comment by xyproto** [comment]
Created a new issue for just the background switching to gray before setting the image here: https://github.com/swaywm/sway/issues/3693

**Comment by Neeek** [comment]
Hello, it looks like this might be back as of 1.5?
on my new system with sway 1:1.5-1, swaybg 1.0-4, and wlroots 0.11.0-1 I am noticing a full hangup that lasts 200ms.

If I open glxgears and run swaymsg output "*" solid_color fill #000000 in another terminal, gears will report 40 dropped frames (running at 144Hz). I checked on my old system running one version behind all the ones I listed above and didn't have this stutter, then I upgraded the packages to current and observed the same behavior.

This is on Arch with the official packages, not AUR or git.
